sql/core/src/test/scala/org/apache/spark/sql/DatasetUnpivotSuite.scala

```diff
@@ -137,48 +138,49 @@ class DatasetUnpivotSuite extends QueryTest
     assert(unpivoted.schema === longSchema)
     checkAnswer(unpivoted, longDataRows)
 
-    val unpivoted2 = wideDataDs.select($"id", $"str1", $"str2")
-      .unpivot(
-        Array($"id"),
-        Array.empty,
-        variableColumnName = "var",
-        valueColumnName = "val")
-    assert(unpivoted2.schema === longSchema)
-    checkAnswer(unpivoted2, longDataRows)
+    val e = intercept[AnalysisException] {
+      wideDataDs.select($"id", $"str1", $"str2")
+        .unpivot(
+          Array($"id"),
+          Array.empty,
+          variableColumnName = "var",
+          valueColumnName = "val")
+    }
+    checkError(
+      exception = e,
+      errorClass = "UNPIVOT_REQUIRES_VALUE_COLUMNS",
```

Comment:
This is arguable. Since value columns can't be `Nil`, it seems reasonable to pass `Array.empty` and expect Spark to infer the value columns.

Reply:
nvm, people should call the other overload if they want Spark to infer value columns.